src/components/floatingButton/index.tsx

```diff
@@ -86,21 +96,32 @@ class FloatingButton extends PureComponent<FloatingButtonProps> {
   getAnimatedStyle = () => {
     return {
       opacity: this.visibleAnimated,
-      transform: [{translateY: this.visibleAnimated.interpolate({
-        inputRange: [0, 1],
-        outputRange: [Constants.screenHeight / 2, 0]
-      })}]
+      transform: [
+        {
+          translateY: this.visibleAnimated.interpolate({
+            inputRange: [0, 1],
+            outputRange: [Constants.screenHeight / 2, 0]
+          })
+        }
+      ]
     };
-  }
+  };
 
   renderButton() {
-    const {bottomMargin, button, secondaryButton, testID} = this.props;
-    const bottom = secondaryButton ? Spacings.s4 : bottomMargin || Spacings.s8;
+    const {bottomMargin, button, secondaryButton, testID, buttonLayout} = this.props;
+
+    const secondaryHorizontal = secondaryButton && buttonLayout === FloatingButtonLayouts.HORIZONTAL;
+    const secondaryVertical = secondaryButton && buttonLayout === FloatingButtonLayouts.VERTICAL;
```

Comment:
This is much better and way more readable. 
I would improve a little more. 
Since you use the same logic in multiple places (renderButton and renderSecondary), I think it'll be better to centralize the logic to a method. 

What we usually do is create a getter method that can be used throughout the component's code. 
It looks like this
```
get isSecondaryVertical() {
    return ...
  }
```

This way, in the  future, if the logic changes, you change it in one place.